modules/ml/src/main/java/org/apache/ignite/ml/inference/IgniteModelStorageUtil.java

```diff
@@ -125,13 +161,30 @@ private static void saveModelDescriptorStorage(String name, UUID mdlId) {
         Ignite ignite = Ignition.ignite();
 
         ModelDescriptorStorage descStorage = new ModelDescriptorStorageFactory().getModelDescriptorStorage(ignite);
-        descStorage.put(name, new ModelDescriptor(
-            name,
+
+        boolean saved = descStorage.putIfAbsent(name, new ModelDescriptor(
+            mdlId.toString(),
             null,
             new ModelSignature(null, null, null),
             new ModelStorageModelReader(IGNITE_MDL_FOLDER + "/" + mdlId),
             new IgniteModelParser<>()
         ));
+
+        if (!saved)
```

Comment:
What about flag force for put or another method "putIfAbsent"?
In my opinion such behavior is implicit. OR you should write about exception in javadoc

Reply:
I've updated javadoc.